lib/puppet/property.rb

```diff
@@ -402,6 +402,7 @@ def property_matches?(current, desired)
   # @param value [Object] the value to format as a string
   # @return [String] a pretty printing string
   def is_to_s(value)
+    value = value[0] if value.is_a?(Array) && value.size == 1
```

Comment:
I'm not completely happy about this either and I'm fine reverting it, but I'd like your opinion on the following before I go ahead.

The reason I added it was that before, the common behavior was to do this:
```
value = [value] unless value.is_a?(Array)
value.join(',')
```
and in a comma separated list it doesn't matter if it's a single value or an array with one element. The output is exactly the same, 'x' is always 'x'. Now, with a unified array representation that always used the [] notation, that's no longer true, 'x' becomes ['x'].